### PR TITLE
service dependencies endpoint

### PR DESCRIPTION
**Pull request by DanielMcSheehy**
added new endpoint for [service dependencies](https://docs.datadoghq.com/api/v1/service-dependencies/). 

result:
```
{
    "wupiupi": {
        "calls": []
    },
    "trugut": {
        "calls": [
            "watto"
        ]
    },
    "tedryn": {
        "calls": []
    },
    "bodo": {
        "calls": [
            "mari",
            "kos",
            "tedryn"
        ]
    },
    "bacta": {
        "calls": []
    },
    "midi": {
        "calls": [
            "watto",
            "phasma"
        ]
    },
    "interstellar": {
        "calls": [
            "effervescing",
            "dooku",
            "watto",
            "kos"
        ]
    },
    "effervescing": {
        "calls": [
            "watto",
            "dooku"
        ]
    },
    "mari": {
        "calls": []
    },
    "dreadnought": {
        "calls": []
    },
    "phasma": {
        "calls": [
            "midi"
        ]
    },
    "kos": {
        "calls": [
            "wupiupi"
        ]
    },
    "dooku": {
        "calls": []
    },
    "effort": {
        "calls": [
            "bodo"
        ]
    },
    "watto": {
        "calls": [
            "midi",
            "trugut",
            "phasma",
            "tedryn"
        ]
    },
    "kamino": {
        "calls": [
            "bacta",
            "kos",
            "dooku",
            "dreadnought",
            "watto",
            "effervescing",
            "tedryn",
            "phasma",
            "wupiupi",
            "midi"
        ]
    }
}
```

### CODE REVIEW

File: service_dependencies.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package datadog
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+type ServiceDependencies struct {
```
Comment (weeyum) [inline service_dependencies.go]: I don't think this belons in here?

**Comment by joeyparsons** [comment]
LGTM

**Review by weeyum — APPROVED** [review]
I think we should remove the effx related struct, but looks good otherwise

File: service_dependencies.go (hunk 1 above)
Comment (DanielMcSheehy) [reply service_dependencies.go]: whoops, forgot to push. Yeah I deleted that, good catch

**Pull request merged**